common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -1094,6 +1094,48 @@ public static Geometry lineInterpolatePoint(Geometry geom, double fraction) {
     return geom.getFactory().createPoint(interPoint);
   }
 
+  public static double perimeter(Geometry geometry, boolean use_spheroid, boolean lenient) {
+    if (use_spheroid && geometry.getSRID() != 4326) {
+      if (lenient) {
+        return 0;
+      } else {
+        throw new IllegalArgumentException(
+            "For spheroidal perimeter calculations, the input geometry must be in the WGS84 CRS (SRID 4326).");
+      }
+    }
+
+    String geomType = geometry.getGeometryType();
+    if (geomType.equalsIgnoreCase(Geometry.TYPENAME_POLYGON)) {
```

Comment:
Can you re-use the logic from ST_LengthSpheroid?

Reply:
The code is there to filter out linear geometries.